datasources/telegram/models.go

```diff
@@ -0,0 +1,243 @@
+package telegram
+
+import (
+	"encoding/json"
+	"github.com/mholt/timeliner"
+	"hash/fnv"
+	"io"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+)
+
+type telegramArchive struct {
+	Profile       telegramProfile       `json:"personal_information"`
+	ChatContainer telegramChatContainer `json:"chats"`
+}
+
+type telegramChatContainer struct {
+	Chats []telegramChat `json:"list"`
+}
+
+type telegramProfile struct {
+	UserID      int    `json:"user_id"`
+	FirstName   string `json:"first_name"`
+	LastName    string `json:"last_name"`
+	PhoneNumber string `json:"phone_number				"`
+	Username    string `json:"username"`
+}
+
+func (item telegramChat) ID() string {
+	return strconv.Itoa(item.ChatID)
+}
+
+func (item telegramChat) Class() timeliner.ItemClass {
+	return timeliner.CLassConversation
+}
+
+//  ------------------------------- Telegram Chat ---------------------------------------------------------
+
+type telegramChat struct {
+	ChatID   int               `json:"id"`
+	Name     string            `json:"name"`
+	ChatType string            `json:"type"`
+	Messages []telegramMessage `json:"messages"`
+
+	ownerID          string
+	ownerName        string
+	firstMessageTime time.Time
+}
+
+func (item telegramChat) Timestamp() time.Time {
+	return item.firstMessageTime
+}
+
+func (item telegramChat) Owner() (id *string, name *string) {
+	return &item.ownerID, &item.ownerName
+}
+
+func (item telegramChat) DataText() (*string, error) {
+	return nil, nil
+}
+
+func (item telegramChat) DataFileName() *string {
+	return nil
+}
+
+func (item telegramChat) DataFileReader() (io.ReadCloser, error) {
+	return nil, nil
+}
+
+func (item telegramChat) DataFileHash() []byte {
+	return nil
+}
+
+func (item telegramChat) DataFileMIMEType() *string {
+	return nil
+}
+
+func (item telegramChat) Metadata() (*timeliner.Metadata, error) {
+	return nil, nil
+}
+
+func (item telegramChat) Location() (*timeliner.Location, error) {
+	return nil, nil
+}
+
+//  ------------------------------- Telegram Message ---------------------------------------------------------
```

Comment:
You could keep these all in one file if you want, but based on how you have this separated, I'd at least suggest just putting these sections/models in different files.